src/main/java/org/deepsymmetry/beatlink/dbserver/ConnectionManager.java

```diff
@@ -280,59 +281,76 @@ private void setPlayerDBServerPort(int player, int port) {
             0x00
     };
 
+    /**
+     * Keeps track of threads currently querying for database server ports, indexed by the address of the device
+     * being queried, so we can make sure we only ask once, even when multiple devices share the same address.
+     * Also allows us to interrupt those threads when we are shutting down.
+     */
+    private final Map<InetAddress,Thread> activeQueryThreads = new ConcurrentHashMap<>();
+
     /**
      * Query a player to determine the port on which its database server is running.
      *
      * @param announcement the device announcement with which we detected a new player on the network.
      */
     private void requestPlayerDBServerPort(DeviceAnnouncement announcement) {
-        // logger.info("Trying to determine database server port for device number " + announcement.getNumber());
-        for (int tries = 0; tries < 4; ++tries) {
+        try {
+            logger.debug("Trying to determine database server port for device number {}", announcement.getDeviceNumber());
```

Comment:
Rather than commenting out the log entry, I just dropped it to DEBUG level, so it normally does nothing, but can be seen if I want debug-level output.